lenskit-core/src/main/java/org/lenskit/rerank/NumCandidates.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * LensKit, an open source recommender systems toolkit.
+ * Copyright 2010-2016 LensKit Contributors.  See CONTRIBUTORS.md.
+ * Work on LensKit has been funded by the National Science Foundation under
+ * grants IIS 05-34939, 08-08692, 08-12148, and 10-17697.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.lenskit.rerank;
+
+import org.grouplens.grapht.annotation.DefaultDouble;
+import org.lenskit.hybrid.RankBlendingItemRecommender;
+import org.lenskit.inject.Parameter;
+
+import javax.inject.Qualifier;
+import java.lang.annotation.*;
+
+/**
+ * Number of candidate items to get from base recommender. Set to negative for no preference.
```

Comment:
What do you think of my proposal: the main recommender always requests full rank, the new item selector (`NextRecommendedItemSelector`?) is then in charge of any possible efficiency / optimality trade-offs. We loose a little in possibly over-requesting the rank, but its a pretty trivial loss for a lot of added flexibility.